transactions/staged_mutation.cxx

```diff
@@ -37,9 +37,8 @@ staged_mutation_queue::add(const staged_mutation& mutation)
 {
     std::lock_guard<std::mutex> lock(mutex_);
     // Can only have one staged mutation per document.
-    auto new_end = std::remove_if(
+    std::remove_if(
       queue_.begin(), queue_.end(), [&mutation](const staged_mutation& item) { return document_ids_equal(item.id(), mutation.id()); });
-    queue_.erase(new_end, queue_.end());
```

Comment:
I should have mentioned this was an experiment, since I didn't completely understand the windows build error.